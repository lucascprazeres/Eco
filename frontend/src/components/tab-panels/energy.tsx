import { CarbonFootprintInput, TabsEnum } from '@eco/models/carbon-footprint'
import { TabPanel } from '@mui/lab'
import { Box, Button, TextField, Typography } from '@mui/material'
import { useFormContext } from 'react-hook-form'

interface EnergyPanelProps {
  onClickNext: () => void
}

export function EnergyPanel({ onClickNext }: EnergyPanelProps) {
  const { register, handleSubmit, formState } =
    useFormContext<CarbonFootprintInput>()

  function handleGoToNextTab() {
    onClickNext()
  }

  const error = formState.errors?.electricityUsageKWhPerMonth?.message

  return (
    <TabPanel
      value={TabsEnum.EnergyUsage}
      style={{ height: '100%', width: '100%' }}
    >
      <form
        onSubmit={handleSubmit(handleGoToNextTab)}
        style={{ display: 'flex', flexDirection: 'column', height: '100%' }}
      >
        <Box
          width="100%"
          height="100%"
          display="flex"
          flexDirection="column"
          gap={4}
        >
          <Typography
            variant="h6"
            fontWeight="bold"
            fontFamily="ubuntu"
            color="secondary.dark"
          >
            Enter your monthly electricity usage
          </Typography>

          <TextField
            {...register('electricityUsageKWhPerMonth', {
              valueAsNumber: true,
              required: {
                value: true,
                message: 'Please insert your electric usage',
              },
            })}
            label="Electricity usage (Kwh/month)"
            type="number"
            fullWidth
            helperText={error}
            error={!!error}
          />

          <Box
            display="flex"
            alignSelf="flex-end"
            alignItems="center"
            justifyContent="flex-end"
            width="100%"
            bottom={0}
          >
            <Button
              type="submit"
              color="primary"
              variant="contained"
              disabled={!!error}
              sx={{ fontWeight: 'bold' }}
            >
              Next
            </Button>
          </Box>
        </Box>
      </form>
    </TabPanel>
  )
}
