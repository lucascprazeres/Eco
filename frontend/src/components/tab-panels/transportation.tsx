import { CarbonFootprintInput } from '@eco/models/carbon-footprint'
import {
  Box,
  Button,
  FormControlLabel,
  Radio,
  RadioGroup,
  TextField,
  Typography,
} from '@mui/material'
import { useController, useFormContext } from 'react-hook-form'

interface TransportationPanelProps {
  onClickNext: () => void
  onGoBack: () => void
}

export function TransportationPanel({
  onClickNext,
  onGoBack,
}: TransportationPanelProps) {
  const { register, handleSubmit, formState, control } =
    useFormContext<CarbonFootprintInput>()

  const { field, fieldState } = useController({
    name: 'transportationFuelType',
    control,
    rules: {
      required: {
        value: true,
        message: 'Please select at least one option',
      },
    },
  })

  function handleGoToNextTab() {
    onClickNext()
  }

  const error = formState.errors?.transportationFuelGallonsPerMonth?.message

  return (
    <form
      onSubmit={handleSubmit(handleGoToNextTab)}
      style={{ display: 'flex', flexDirection: 'column', height: '100%' }}
    >
      <Box
        width="100%"
        height="100%"
        display="flex"
        flexDirection="column"
        gap={4}
      >
        <Box>
          <Typography variant="h6" fontWeight="bold" fontFamily="ubuntu">
            What type of fuel does your vehicle use?
          </Typography>

          <RadioGroup
            name="transporationFuelType"
            value={field.value}
            onChange={field.onChange}
          >
            <Box display="flex" alignItems="center" gap={1}>
              <FormControlLabel
                value="gasoline"
                label="Gasoline"
                control={<Radio />}
              />
              <FormControlLabel
                label="Diesel"
                value="diesel"
                control={<Radio />}
              />
              <FormControlLabel
                label="Ethanol"
                value="ethanol"
                control={<Radio />}
              />
            </Box>
          </RadioGroup>

          <Typography variant="caption" color="error">
            {fieldState.error?.message}
          </Typography>
        </Box>

        <Typography variant="h6" fontWeight="bold" fontFamily="ubuntu">
          How many gallons of fuel do you use in a month?
        </Typography>

        <TextField
          {...register('transportationFuelGallonsPerMonth', {
            valueAsNumber: true,
            required: {
              value: true,
              message: 'Please insert your fuel usage (or 0 if none)',
            },
          })}
          label="Fuel (gallons)"
          type="number"
          style={{ width: '100%' }}
          helperText={error}
          error={!!error}
        />

        <Box
          display="flex"
          alignSelf="flex-end"
          alignItems="center"
          justifyContent="space-between"
          width="100%"
          bottom={0}
        >
          <Button variant="outlined" onClick={onGoBack}>
            Go back
          </Button>
          <Button
            type="submit"
            color="primary"
            variant="contained"
            disabled={!!error}
          >
            Next
          </Button>
        </Box>
      </Box>
    </form>
  )
}
