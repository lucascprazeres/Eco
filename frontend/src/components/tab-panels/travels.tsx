import { RESULTS_PAGE } from '@eco/constants/routes'
import { CarbonFootprintInput } from '@eco/models/carbon-footprint'
import { Box, Button, TextField, Typography } from '@mui/material'
import { useRouter } from 'next/router'
import { useCallback } from 'react'
import { useFormContext } from 'react-hook-form'

interface TravelPanelProps {
  onGoBack: () => void
  onSubmit: (data: CarbonFootprintInput) => Promise<void>
}

export function TravelPanel({ onGoBack, onSubmit }: TravelPanelProps) {
  const router = useRouter()
  const { register, handleSubmit, formState } =
    useFormContext<CarbonFootprintInput>()

  const handleLastSubmit = useCallback(
    async (data: CarbonFootprintInput) => {
      try {
        await onSubmit(data)
        router.push(RESULTS_PAGE)
      } catch (err) {
        console.log(err)
      }
    },
    [router, onSubmit],
  )

  const travelsPerYearError = formState.errors?.airTravelsPerYear?.message

  return (
    <form
      onSubmit={handleSubmit(handleLastSubmit)}
      style={{ display: 'flex', flexDirection: 'column', height: '100%' }}
    >
      <Box
        width="100%"
        height="100%"
        display="flex"
        flexDirection="column"
        gap={4}
      >
        <Typography
          variant="h6"
          fontWeight="bold"
          fontFamily="ubuntu"
          color="secondary.dark"
        >
          How many flights do you take per year?
        </Typography>

        <TextField
          {...register('airTravelsPerYear', {
            valueAsNumber: true,
            required: {
              value: true,
              message:
                'Please insert how many flights do you take (or 0 if none)',
            },
          })}
          label="Flights (per year)"
          type="number"
          fullWidth
          helperText={travelsPerYearError}
          error={!!travelsPerYearError}
        />

        <Box
          display="flex"
          alignSelf="flex-end"
          alignItems="center"
          justifyContent="space-between"
          width="100%"
          bottom={0}
        >
          <Button
            variant="outlined"
            onClick={onGoBack}
            sx={{ fontWeight: 'bold' }}
          >
            Go back
          </Button>
          <Button
            type="submit"
            color="primary"
            variant="contained"
            disabled={!!travelsPerYearError}
            sx={{ fontWeight: 'bold' }}
          >
            Next
          </Button>
        </Box>
      </Box>
    </form>
  )
}
