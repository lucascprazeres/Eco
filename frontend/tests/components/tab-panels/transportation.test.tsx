import { describe, it, expect, vi, Mock, afterEach } from 'vitest'
import { render, screen, fireEvent, cleanup } from '@testing-library/react'
import { TransportationPanel } from '@eco/components/tab-panels/transportation'
import { useFormContext, useController } from 'react-hook-form'
import {
  useControllerMockReturn,
  useFormContextMockReturn,
} from '../../mocks/react-hook-form'

vi.mock('react-hook-form', () => ({
  useFormContext: vi.fn(() => useFormContextMockReturn),
  useController: vi.fn(() => useControllerMockReturn),
}))

describe('TransportationPanel', () => {
  afterEach(() => {
    cleanup()
  })

  it('should be able to render correctly', () => {
    // arrange
    const mockOnClickNext = vi.fn()
    const mockOnGoBack = vi.fn()

    // act
    const result = render(
      <TransportationPanel
        onClickNext={mockOnClickNext}
        onGoBack={mockOnGoBack}
      />,
    )

    const fuelTypeTitle = screen.getByText(
      /What type of fuel does your vehicle use\?/i,
    )
    const fuelUsageTitle = screen.getByText(
      /How many gallons of fuel do you use in a month\?/i,
    )
    const fuelInput = screen.getByLabelText(/Fuel \(gallons\)/i)
    const nextButton = screen.getByRole('button', { name: /Next/i })
    const backButton = screen.getByRole('button', { name: /Go back/i })

    // assert
    expect(fuelTypeTitle).toBeInTheDocument()
    expect(fuelUsageTitle).toBeInTheDocument()
    expect(fuelInput).toBeInTheDocument()
    expect(nextButton).toBeInTheDocument()
    expect(backButton).toBeInTheDocument()
    expect(nextButton).not.toBeDisabled()
    expect(result).toMatchSnapshot()
  })

  it('should be able to display validation error for fuel type', () => {
    // arrange
    // eslint-disable-next-line prettier/prettier
    (useController as Mock).mockReturnValueOnce({
      ...useControllerMockReturn,
      fieldState: { error: { message: 'Please select at least one option' } },
    })

    // act
    render(<TransportationPanel onClickNext={vi.fn()} onGoBack={vi.fn()} />)

    const helperText = screen.getByText(/Please select at least one option/i)

    // assert
    expect(helperText).toBeInTheDocument()
  })

  it('should display validation error for fuel usage', () => {
    // arrange
    // eslint-disable-next-line prettier/prettier
    (useFormContext as Mock).mockReturnValueOnce({
      ...useFormContextMockReturn,
      formState: {
        errors: {
          transportationFuelGallonsPerMonth: {
            message: 'Please insert your fuel usage (or 0 if none)',
          },
        },
      },
    })

    // act
    render(<TransportationPanel onClickNext={vi.fn()} onGoBack={vi.fn()} />)

    const helperText = screen.getByText(
      /Please insert your fuel usage \(or 0 if none\)/i,
    )

    // assert
    expect(helperText).toBeInTheDocument()
  })

  it('should call onClickNext when form is submitted without errors', () => {
    // arrange
    const mockOnClickNext = vi.fn()

    // act
    render(
      <TransportationPanel onClickNext={mockOnClickNext} onGoBack={vi.fn()} />,
    )

    const nextButton = screen.getByRole('button', { name: /Next/i })
    fireEvent.submit(nextButton)

    // assert
    expect(mockOnClickNext).toHaveBeenCalledTimes(1)
  })

  it('should call onGoBack when back button is clicked', () => {
    // arrange
    const mockOnGoBack = vi.fn()

    // act
    render(
      <TransportationPanel onClickNext={vi.fn()} onGoBack={mockOnGoBack} />,
    )

    const backButton = screen.getByRole('button', { name: /Go back/i })
    fireEvent.click(backButton)

    // assert
    expect(mockOnGoBack).toHaveBeenCalledTimes(1)
  })
})
