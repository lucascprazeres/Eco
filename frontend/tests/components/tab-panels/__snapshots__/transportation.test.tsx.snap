// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TransportationPanel > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        style="display: flex; flex-direction: column; height: 100%;"
      >
        <div
          class="MuiBox-root css-nf1kcv"
        >
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-b5iqwz-MuiTypography-root"
            >
              What type of fuel does your vehicle use?
            </h6>
            <div
              class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
              role="radiogroup"
            >
              <div
                class="MuiBox-root css-axw7ok"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-j8yymo"
                      name="transporationFuelType"
                      type="radio"
                      value="gasoline"
                    />
                    <span
                      class="css-1qiat4j"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    Gasoline
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-j8yymo"
                      name="transporationFuelType"
                      type="radio"
                      value="diesel"
                    />
                    <span
                      class="css-1qiat4j"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    Diesel
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-j8yymo"
                      name="transporationFuelType"
                      type="radio"
                      value="ethanol"
                    />
                    <span
                      class="css-1qiat4j"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    Ethanol
                  </span>
                </label>
              </div>
            </div>
            <span
              class="MuiTypography-root MuiTypography-caption css-8gz0an-MuiTypography-root"
            />
          </div>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-b5iqwz-MuiTypography-root"
          >
            How many gallons of fuel do you use in a month?
          </h6>
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            style="width: 100%;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Fuel (gallons)
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                type="number"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-81qg8w"
                >
                  <span>
                    Fuel (gallons)
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-14i6vh8"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-uah78b-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Go back
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r9p8f3-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Next
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      style="display: flex; flex-direction: column; height: 100%;"
    >
      <div
        class="MuiBox-root css-nf1kcv"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-b5iqwz-MuiTypography-root"
          >
            What type of fuel does your vehicle use?
          </h6>
          <div
            class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
            role="radiogroup"
          >
            <div
              class="MuiBox-root css-axw7ok"
            >
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
              >
                <span
                  class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-j8yymo"
                    name="transporationFuelType"
                    type="radio"
                    value="gasoline"
                  />
                  <span
                    class="css-1qiat4j"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                >
                  Gasoline
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
              >
                <span
                  class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-j8yymo"
                    name="transporationFuelType"
                    type="radio"
                    value="diesel"
                  />
                  <span
                    class="css-1qiat4j"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                >
                  Diesel
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
              >
                <span
                  class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-j8yymo"
                    name="transporationFuelType"
                    type="radio"
                    value="ethanol"
                  />
                  <span
                    class="css-1qiat4j"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                      data-testid="RadioButtonUncheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                      data-testid="RadioButtonCheckedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                >
                  Ethanol
                </span>
              </label>
            </div>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-8gz0an-MuiTypography-root"
          />
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-b5iqwz-MuiTypography-root"
        >
          How many gallons of fuel do you use in a month?
        </h6>
        <div
          class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
          style="width: 100%;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            Fuel (gallons)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-81qg8w"
              >
                <span>
                  Fuel (gallons)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-14i6vh8"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-uah78b-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Go back
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r9p8f3-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Next
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
